Mage/src/main/java/mage/cards/ExpansionSet.java

```diff
@@ -295,6 +292,29 @@ public void shuffleCollator() {
                 .collect(Collectors.toList());
     }
 
+    public void openBoosterBox() {
+        // nothing to do if not using collator
+        if (boosterCollator == null) {
+            return;
+        }
+        // don't open a new box if the old one hasn't been used yet
+        if (boosterBox.size() >= boosterBoxSize) {
+            return;
```

Comment:
this seems to be at odds with all the other code calling this function, given that we get a new box opened if and only if there aren't any packs left in the current one

Reply:
This check is merely a performance optimization so that in sealed deck we don't generate six boxes in a row and throw away the first five.